### ISSUE TITLE
Java 14: jaxb.Persistence: Unsupported class file major version 58

### ISSUE THREAD

**Issue by barchetta** [oracle/helidon#2080]
## Environment Details
* Helidon Version: 2.0.0
* Helidon MP
* JDK version: 14
* OS: MacOS
----------

## Problem Description

We have seen an error when building the database-mp archetype with JDK 14.

Also see the error with `io.helidon.tests.integration:helidon-tests-native-image-mp-2` when built and run with JDK 14 (see below). These work fine with JDK 11.

```
[ERROR] Failed to execute goal org.hibernate.orm.tooling:hibernate-enhance-maven-plugin:5.4.6.Final:enhance (default) on project helidon-tests-native-image-mp-2: Unable to enhance class: Persistence.class: Failed to enhance class io.helidon.tests.integration.nativeimage.mp2.jaxb.Persistence: Unsupported class file major version 58 -> [Help 1]
```

**Comment by ljnelson** [comment]
(Blindly guessing that version [5.4.18](https://search.maven.org/classic/#artifactdetails%7Corg.hibernate.orm.tooling%7Chibernate-enhance-maven-plugin%7C5.4.18.Final%7Cjar) will fix it.)

**Comment by barchetta** [comment]
I did a quick test and Laird guessed well. 5.4.18 appears to fix this.

Also when we fix this, we should update the archetype so that it uses the following properties for the plugin versions (these are already defined in the application parent pom):

```
                <version>${version.plugin.jaxb2}</version>
                <version>${version.plugin.hibernate.enhance}</version>
```

**Comment by ljnelson** [comment]
The one remaining question is: should/must the plugin version track the "main" Hibernate version?  Probably.

**Comment by barchetta** [comment]
Fixed by PR #2212
